<table align="center" width=100%>
    <tr>
        <td width="10%">
            <img src="in_class.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Inclass-Lab  <br>(Week 3)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

## Table of Content

1. **[Conditional Statements](#cond)**
2. **[Python Flow Control](#flow_control)**
3. **[List Comprehension](#lc)**



<a id="cond"> </a>
## 1. Conditional Statements

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Write a program to check whether a given number is multiple of 7 (use if statement)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
a=int(input("Enter the number"))
print("Multiply of 7" if(a%7==0) else "Not a multiply of 7")

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. Write a code to print the maximum of two numbers (use if-else)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
a=int(input("Enter the number A :"))
b=int(input("Enter the number B :"))
print("A is maximum" if(a>b) else "B is maximun")

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Write a program to check whether a number is positive, negative or zero (use if-elif) </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
a=int(input("Enter the number A :"))
if(a>0):
    print("A is Positive" )
elif (a<0):
    print("A is Negative")
else: print("A is zero")

### Let's begin with some hands-on practice exercises

<a id= "flow_control"> </a>
## 2. Python Flow Control

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Write a code to print table of 5 using assignment operand +=.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
num=5
for i in range(0,11):
    print(num)
    num+=5
    

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. Write a program for generating a fibonacci series starting with 0 and 1 for the next 10 values using a while loop. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
# type your code here  
a,b=0,1
i=0
while i<=10:
    c=a+b
    a=b
    b=c
    i=i+1
    print(c ,end=" ")

1 2 3 5 8 13 21 34 55 89 144 

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. Write a program to check whether a number is greater than the other using if statement (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [12]:
# type your code here  
a=int(input("Enter the number A :"))
b=int(input("Enter the number B :"))
print(str(a)+' is maximum' if(a>b) else str(b)+' is maximum')

Enter the number A :23
Enter the number B :56
56 is maximum


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Write a code to check whether a number is divisible by 7 or not (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
a=int(input("Enter the number"))
print("Multiply of 7" if(a%7==0) else "Not a multiply of 7")

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Write a code to find factorial of a number (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
# type your code here
fact=1
a=int(input("Enter the number : "))
if a==0:
    print("Factorial not available for 0")
elif a==1:
    print("Factorial of 1 is 1")
else:
    for i in range (1,a):
        fact=fact*i
    print(fact)

Enter the number : 5
24


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Write a program to check whether a number is prime or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [14]:
# type your code here
a=int(input("Enter the number : "))
for i in range(2,a):
    if(a%i==0):
        print("Not a prime number")
        break
else :
        print("Prime number")

Enter the number : 13
Prime number


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. Write a program to check whether two numbers are amicable or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
a=int(input("Enter the number : "))
b=int(input("Enter the number : "))

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. Reverse string using a for loop (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [23]:
# type your code here
a=input("Enter the variable")
str = ""
for i in a:
    str=i+str
print(str)

Enter the variableasdd
ddsa


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. Write a code to find the average of given numbers (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
sum=0
for i in range(3):
    sum+=int(input("Enter the character :"))
print(sum/3)
    

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. Write a program to find the area of a circle for a given radius (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
# type your code here
a=int(input("Enter the character :"))
print(3.14*a**2)

Enter the character :3
28.26


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. Write a code to find the simple interest (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
# type your code here
p=int(input("Enter the character :"))
n=int(input("Enter the character :"))
r=int(input("Enter the character :"))
print((p*n*r)/100)

Enter the character :600
Enter the character :5
Enter the character :7
210.0


<a id= "lc"> </a>
## 3. List Comprehension

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Use list comprehension to obtain the squre root of first 10 natural numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [12]:
##### type your code here
a=[]
for i in range(1,11):
    a.append(i**2)
print(a)


[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]


In [8]:

a=list( a*a for a in range(1,11))
a

[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Use list comprehension to find the used vowels in a given sentence and/or word (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [9]:
# type your code here
sen="ashwin"
a=[]
for i in sen:
    if i in ('aeiou'):
        a.append(i)
print(a)

['a', 'i']


In [11]:
sen="ashwin"
list( a for a in sen if a in ('aeiou'))

['a', 'i']

# <table>
<tr>
    <th>Key</th>
    <td>1</td>
    <td>2</td>
    <td>3</td>
    <td>4</td> 
    <td>5</td> 
</tr>
<tr>
    <th>Name</th>
    <td>Aman</td>
    <td>Mohit</td>
    <td>Guari</td>
    <td>Imran</td>
    <td>Roma</td
     </tr>
<tr>
    <th>Marks</th>
    <td>24</td>
    <td>25</td>
    <td>26</td>
    <td>24</td>
    <td>27</td>
   </tr>
</table>

In [21]:
# type your code here
a={'key': [1,2,3,4,5],'Name':['Aman','Mohit','Guari','Imran','Roma'],'Marks':[24,25,26,24,27]}
print(a)

{'key': [1, 2, 3, 4, 5], 'Name': ['Aman', 'Mohit', 'Guari', 'Imran', 'Roma'], 'Marks': [24, 25, 26, 24, 27]}


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. Use list comprehension to find even and odd numbers from first 20 whole numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [23]:
# type your code here        
a={i : 'Even' if(i%2==0) else "Odd" for i in range(1,20)  }
print(a)

{1: 'Odd', 2: 'Even', 3: 'Odd', 4: 'Even', 5: 'Odd', 6: 'Even', 7: 'Odd', 8: 'Even', 9: 'Odd', 10: 'Even', 11: 'Odd', 12: 'Even', 13: 'Odd', 14: 'Even', 15: 'Odd', 16: 'Even', 17: 'Odd', 18: 'Even', 19: 'Odd'}


In [15]:
list('Even' if (a%2==0) else 'odd' for a in range(0,20) )

['Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd',
 'Even',
 'odd']

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. Use list comprehension to print numbers divisible  by 2 and 3 in between 1 and 100. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [29]:
# type your code here
a=[]
for i in range(1,50):
    if (i%2==0 or i%3==0):
        a.append(i)
print(a)

[2, 3, 4, 6, 8, 9, 10, 12, 14, 15, 16, 18, 20, 21, 22, 24, 26, 27, 28, 30, 32, 33, 34, 36, 38, 39, 40, 42, 44, 45, 46, 48]


In [22]:
list( a for a in range(1,100) if((a%3==0) and (a%2==0)) )

[6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Use list comprehension to create a dictionary such that its keys are numbers from 1 to 10 and values are the corresponding to that key are its cube.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [32]:
# type your code here

a={i:i**3 for i in range(1,11)}
print(a)

{1: 1, 2: 8, 3: 27, 4: 64, 5: 125, 6: 216, 7: 343, 8: 512, 9: 729, 10: 1000}


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>21. Use list comprehension to extract numbers from a string </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [45]:
# type your code here
a="asd123rt"
b=[int(i) for i in a  if i.isdigit()]
print(b)

[1, 2, 3]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>22. Use list comprehension to print table of 11 to 20 as shown below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
<tr>
    <td>11</td><td>22</td><td>33</td><td>44</td><td>55</td><td>66</td><td>77</td><td>88</td><td>99</td><td>110</td> 
</tr>
    
<tr>
    <td>12</td><td>24</td><td>36</td><td>48</td><td>60</td><td>72</td><td>84</td><td>96</td><td>108</td><td>120</td> 
</tr>

<tr>
    <td>13</td><td>36</td><td>39</td><td>52</td><td>65</td><td>78</td><td>91</td><td>104</td><td>117</td><td>130</td> 
</tr> 

<tr>
     <td>14</td><td>28</td><td>42</td><td>56</td><td>70</td> <td>84</td><td>98</td><td>112</td><td>126</td><td>140</td> 
</tr>

<tr>
     <td>15</td><td>30</td><td>45</td><td>60</td><td>75</td><td>90</td><td>105</td><td>120</td><td>135</td><td>150</td> 
</tr>

<tr>
     <td>16</td><td>32</td><td>48</td><td>64</td><td>80</td><td>96</td><td>112</td><td>128</td><td>144</td><td>160</td> 
</tr> 

<tr>
     <td>17</td><td>34</td><td>51</td><td>68</td><td>85</td><td>102</td><td>119</td><td>136</td><td>153</td><td>170</td> 
</tr> 

<tr>
     <td>18</td><td>36</td><td>54</td><td>72</td><td>90</td> <td>108</td><td>126</td><td>144</td><td>162</td><td>180</td> 
</tr>

<tr>
 <td>19</td><td>38</td><td>57</td><td>76</td><td>95</td><td>114</td><td>133</td><td>152</td><td>171</td><td>190</td> 
</tr>

<tr>
     <td>20</td><td>40</td><td>60</td><td>80</td><td>100</td><td>120</td><td>140</td><td>160</td><td>180</td><td>200</td> 
<tr>    
    
</table>

In [53]:
# type your code here

d=[[i*j for j in range(1,11)] for i in range(11,21)]
d

[[11, 22, 33, 44, 55, 66, 77, 88, 99, 110],
 [12, 24, 36, 48, 60, 72, 84, 96, 108, 120],
 [13, 26, 39, 52, 65, 78, 91, 104, 117, 130],
 [14, 28, 42, 56, 70, 84, 98, 112, 126, 140],
 [15, 30, 45, 60, 75, 90, 105, 120, 135, 150],
 [16, 32, 48, 64, 80, 96, 112, 128, 144, 160],
 [17, 34, 51, 68, 85, 102, 119, 136, 153, 170],
 [18, 36, 54, 72, 90, 108, 126, 144, 162, 180],
 [19, 38, 57, 76, 95, 114, 133, 152, 171, 190],
 [20, 40, 60, 80, 100, 120, 140, 160, 180, 200]]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>23. Use list comprehension to find transpose of the matrix given below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
    <tr>
        <td>1</td>
        <td>2</td>
    </tr>
    <tr>
        <td>3</td>
        <td>4</td>
    </tr>
    <tr>
        <td>5</td>
        <td>6</td>
    </tr>
    <tr>
        <td>7</td>
        <td>8</td>
    </tr>
    </table>

In [60]:
# type your code here
b=[[1,2],[3,4],[5,6],[7,8]]
for i in range (0,3):
    for j in range (0,1):
        a[i][j]=b[j][i]


IndexError: list index out of range

In [63]:
b=[[1,2],[3,4]]
for i in range (0,1):
    for j in range (0,1):
        a[i][j]=b[j][i]
a

[[1, 3], [2, 4], [5, 6], [7, 8]]